src/index.ts

```diff
@@ -263,7 +271,14 @@ class VirtualModulesPlugin {
       const fts = compiler.fileTimestamps as any;
       if (virtualFiles && fts && typeof fts.set === 'function') {
         Object.keys(virtualFiles).forEach((file) => {
-          fts.set(file, +virtualFiles[file].stats.mtime);
+          const mtime = +virtualFiles[file].stats.mtime;
+          if (mtime) applyMtime(mtime);
+          // webpack 5.58.2
+          fts.set(file, {
+              accuracy: 0,
+              safeTime: mtime ? mtime + FS_ACCURACY : Infinity,
+              timestamp: mtime
+          });
```

Comment:
The [Webpack 5 watch graph](https://webpack.js.org/contribute/plugin-patterns/#monitoring-the-watch-graph) behavior has apparently changed as of Webpack 5 breaking this library which is the root of the issue preventing compatibility.

When this function is fired (a file is created or updated) the watch graph needs to be informed so it's included into the build in watch-mode. Webpack does this basically by comparing timestamps (makes sense). I haven't been able to find for-sure what `safeTime` is referring to, but it seems to be used here as "nearest time there is no update" or something like that.

Reply:
Good find, thanks!